projects/pgai/pgai/semantic_catalog/render.py

```diff
@@ -0,0 +1,234 @@
+from pathlib import Path
+
+import psycopg
+from jinja2 import Environment, FileSystemLoader
+from psycopg.sql import SQL, Identifier, Literal
+
+from pgai.semantic_catalog.models import (
+    Fact,
+    ObjectDescription,
+    Procedure,
+    SQLExample,
+    Table,
+    View,
+)
+
+template_dir = Path(__file__).parent.joinpath("templates")
+env = Environment(loader=FileSystemLoader(template_dir))
+
+
+def render_table(table: Table) -> str:
+    template = env.get_template("table.j2")
+    return template.render(table=table)
+
+
+def render_tables(tables: list[Table]) -> str:
+    return "\n\n".join(map(render_table, tables)).strip()
+
+
+def render_view(view: View) -> str:
+    template = env.get_template("view.j2")
+    return template.render(view=view)
+
+
+def render_views(views: list[View]) -> str:
+    return "\n\n".join(map(render_view, views)).strip()
+
+
+def render_procedure(proc: Procedure) -> str:
+    template = env.get_template("procedure.j2")
+    return template.render(proc=proc)
+
+
+def render_procedures(procedures: list[Procedure]) -> str:
+    return "\n\n".join(map(render_procedure, procedures)).strip()
+
+
+def render_objects(objects: list[Table | View | Procedure]) -> str:
+    output = ""
+    for obj in objects:
+        match obj:
+            case Table():
+                output += render_table(obj)
+            case View():
+                output += render_view(obj)
+            case Procedure():
+                output += render_procedure(obj)
+        output += "\n\n"
+    return output.rstrip()
+
+
+def render_fact(fact: Fact) -> str:
+    template = env.get_template("fact.j2")
+    return template.render(fact=fact)
+
+
+def render_sql_example(example: SQLExample) -> str:
+    template = env.get_template("sql_example.j2")
+    return template.render(example=example)
+
+
+def render_description_to_sql(
```

Comment:
I would call this something other than "render". rendering db obj-> text seems a different thing than rendering the "migration_file".

Reply:
any suggestions? To me, it's just another rendering format.

Reply:
I would call this output_sql_statements